perl_lib/PHEDEX/Transfer/Backend/Job.pm

```diff
@@ -361,26 +361,30 @@ sub PrepareJson
   #print "Using temporary file $file \n";
   $self->{JSONCOPYJOB} = $file;
 
+  # be carefull undef are traslate to null strings in json, if they are not present, it is better to remove them
   my @jobfiles = map(
                      { sources => [ $_->{SOURCE} ],
                        destinations => [ $_->{DESTINATION} ],
-                       metadata => undef,
+  #                     metadata => undef,
                        filesize => $_->{FILESIZE},
                        checksums => ( $_->{CHECKSUM_TYPE} && $_->{CHECKSUM_VAL} ) ? $_->{CHECKSUM_TYPE}.':'.$_->{CHECKSUM_VAL} : undef,
                      }, values %{ $self->{FILES} } 
                     );
 
+  #some clean up to avoid mis-interpretation of null values
+  for my $i (0 .. $#jobfiles) { delete $jobfiles[$i]{checksums} unless (defined $jobfiles[$i]{checksums}); }
```

Comment:
Proposing this instead for conciseness, it should be equivalent right?

foreach (@jobfiles) { delete $_->{checksums} unless (defined $_->{checksums}); }